catalog/plugin/catalog-plugin-clientinfo/src/main/java/org/codice/ddf/catalog/plugin/clientinfo/ClientInfoSetupPlugin.java

```diff
@@ -0,0 +1,112 @@
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PreAuthorizationPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+/**
+ * Injects client-specific information, if any exists for the current thread, into request properties.
+ */
+public class ClientInfoSetupPlugin implements PreAuthorizationPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientInfoSetupPlugin.class);
+
+    private static final String CLIENT_INFO_KEY = "client-info";
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest input,
+            Map<String, Metacard> existingMetacards) throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    @Override
+    public DeleteRequest processPreDelete(DeleteRequest input) throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    @Override
+    public DeleteResponse processPostDelete(DeleteResponse input) throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    @Override
+    public QueryRequest processPreQuery(QueryRequest input) throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    @Override
+    public QueryResponse processPostQuery(QueryResponse input) throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    @Override
+    public ResourceRequest processPreResource(ResourceRequest input)
+            throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    @Override
+    public ResourceResponse processPostResource(ResourceResponse input, Metacard metacard)
+            throws StopProcessingException {
+        injectClientInfo(input.getProperties());
+        return input;
+    }
+
+    /**
+     * Assuming a client info map was added to the shiro {@link ThreadContext}, we retrieve
+     * the value and put it into the request properties. The corresponding CXF filter in
+     * {@code platform-filter-clientinfo} is responsible for removing the data to prevent leak.
+     *
+     * @param properties the request properties for the catalog framework.
+     */
+    private void injectClientInfo(Map<String, Serializable> properties) {
+        Object clientInfo = ThreadContext.get(CLIENT_INFO_KEY);
+        if (clientInfo == null) {
```

Comment:
We should also check that `clientInfo instanceof Serializable` to make sure we don't blow up when we cast it on line 109.